pkg/bgpv1/gobgp/reconcile.go

```diff
@@ -12,57 +12,48 @@ import (
 	"github.com/sirupsen/logrus"
 
 	"github.com/cilium/cilium/pkg/bgpv1/agent"
+	"github.com/cilium/cilium/pkg/hive/cell"
 	v2alpha1api "github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2alpha1"
 )
 
-// ConfigReconcilerFunc is a function signature for reconciling a particular aspect
+// ConfigReconciler is a interface for reconciling a particular aspect
 // of an old and new *v2alpha1api.CiliumBGPVirtualRouter
-//
-// If the `Config` field in `sc` is nil the reconciler should unconditionally
-// perform the reconciliation actions, as no previous configuration is present.
-type ConfigReconcilerFunc func(ctx context.Context, m *BGPRouterManager, sc *ServerWithConfig, newc *v2alpha1api.CiliumBGPVirtualRouter, cstate *agent.ControlPlaneState) error
+type ConfigReconciler interface {
+	// Priority is used to determine the order in which reconcilers are called. Reconcilers are called from lowest to
+	// highest.
+	Priority() int
```

Comment:
> For instance, what happens if someone wants to introduce an out-of-tree or plugable ConfigReconciler, and wants to place its execution somewhere between existing ones.

My current answer to that issue is that all existing reconcilers have a spacing of 10, so 9 other reconcilers should be able to order between each existing one.

> There would be no way to do this, previous, at least the slice of ConfigReconcilerFuncs could be reallocated or reordered, but now each Reconciler is defining its order concretely. Is this desired?

Come to think of this, the whole ordering is here since some reconcilers build atop config make by others. But it doesn't matter if the PodCIDR or SvcLB reconciler runs first for example. Perhaps each reconciler should be able to say that is should run after one or more other reconcilers. A bit like the SystemD approach. Then we can sort based on those dependencies without predefined priority and its associated downfalls.